test/runtests.jl

```diff
@@ -1090,17 +1091,26 @@ Adapt.adapt_storage(::ArrayConverter, xs::AbstractArray) = convert(Array, xs)
     @test t.b.d isa Array
 end
 
-struct MyArray{T,N} <: AbstractArray{T,N}
-    A::Array{T,N}
+for S in (1, 2, 3)
```

Comment:
This is a bit hard to follow. It may be best to explain something like the comments below (on the role of types `MyArray1, MyArray2, MyArray3` on testing different possibilities).